### PR TITLE
Improve refreshing on the dining screen

### PR DESCRIPTION
**Pull request by anli5005**
This PR:

1. Stops refreshes from being cancelled when the dining analytics screen is entered (fixing that weird dining analytics refresh cancellation issue)
2. Adds :sparkles: ***animation*** :sparkles:

https://user-images.githubusercontent.com/7005789/213819740-5efd7731-cf6e-4987-8543-b6c532db4bf6.mov

Blame @JHawk0224 for this. In other news, please send help

### CODE REVIEW

**Comment by julianweng** [comment]
Good animation!

**Comment by anli5005** [comment]
I haven't tested this with #447 yet, so maybe that's a thing we should do

**Comment by JHawk0224** [comment]
I think it's good as long as we actually want to add this refresh button

File: PennMobile/Dining/SwiftUI/Views/Venue/DiningVenueView.swift [hunk 1]
```diff
@@ -29,25 +79,29 @@ struct DiningVenueView: View {
...
 
     let name: String
-    var refreshButton = false
+    var refreshConfiguration = RefreshConfiguration.noRefresh
     @State var didError = false
     @State var showMissingDiningTokenAlert = false
     @State var showDiningLoginView = false
+    @State var buttonAngle: Angle = .zero
     @Environment(\.presentationMode) var presentationMode
+    @Environment(\.accessibilityReduceMotion) var reduceMotion
```
Comment (aantenberg) [inline PennMobile/Dining/SwiftUI/Views/Venue/DiningVenueView.swift:104]: Nice accessibility fix - good catch!

**Review by aantenberg — APPROVED** [review]
Looks good to me - nice job

**Pull request merged**